### PR TITLE
Fill is missing fields of index.md.

### LINKED ISSUE #9: Update index.md
`index.md` is the entry point for the course, showing contents, syllabus, learning outcomes and schedule. This should be updated with the correct information for the course, although probably it makes sense to do it once we have an initial draft of the material.

### PR DESCRIPTION
**Pull request by jamesturner246**
Adds the remaining parts of `index.md`.

Fixes #9

### CODE REVIEW

File: index.md [hunk 1]
```diff
@@ -4,15 +4,38 @@ root: .  # Is the only page that doesn't follow the pattern /:path/index.html
...
 
-TBC
+This course teaches the fundamental concepts and skills required to design effective
+test suites for modern C++ software.
+
+The consequences of writing buggy software can range from the mild to the catastrophic,
+but one thing remains invariant: many bugs can be discovered and fixed quickly with a
+well designed and maintained suite of tests. Well tested code inspires confidence in
+stakeholders, and the absence of tests is often viewed as an unacceptable liability,
+particularly in scientific and engineering applications.
```
Comment (dalonsoa) [inline index.md:14]: I love this paragraph ❤️

**Review by dalonsoa — APPROVED** [review]
Perfect!

**Review by lkr-ind — APPROVED** [review]

File: index.md (hunk 1 above)
Comment (alexdewar) [reply index.md:14]: Same :smile:

File: index.md [hunk 2]
```diff
@@ -4,15 +4,38 @@ root: .  # Is the only page that doesn't follow the pattern /:path/index.html
...
+To ensure that your software continues to meet the increasingly stringent expectations
+of users and stakeholders in a sustainable and future-proof way, we will learn to
+implement modern software testing concepts, such as fixtures and parametrised testing using
+the popular [GoogleTest](https://github.com/google/googletest) C++ testing framework.
+
 
 ## Syllabus
 
-TBC
+- Introduction and motivation of software testing.
```
Comment (alexdewar) [inline index.md]: Does "...for software testing" sound better?

```suggestion
- Introduction and motivation for software testing.
```

**Review by alexdewar — APPROVED** [review]
Well written. LGTM :smile:

**Pull request merged**